File: apps/passenger-app/lib/features/ride/services/passenger_ride_service.dart
import '../../../core/api/api_client.dart';

class PassengerRideService {
  final ApiClient apiClient;

  PassengerRideService(this.apiClient);

  Future<Map<String, dynamic>> requestRide({
    required double pickupLat,
    required double pickupLng,
    required String pickupAddress,
    required double dropoffLat,
    required double dropoffLng,
    required String dropoffAddress,
    String serviceType = 'ECONOMY',
  }) async {
    try {
      final response = await apiClient.dio.post('/passenger/rides', data: {
        'pickupLat': pickupLat,
        'pickupLng': pickupLng,
        'pickupAddress': pickupAddress,
        'dropoffLat': dropoffLat,
        'dropoffLng': dropoffLng,
        'dropoffAddress': dropoffAddress,
        'serviceType': serviceType,
      });
      return response.data;
    } catch (e) {
      rethrow;
    }
  }

  Future<Map<String, dynamic>> getRideStatus(String rideId) async {
    try {
      final response = await apiClient.dio.get('/passenger/rides/$rideId');
      return response.data;
    } catch (e) {
      rethrow;
    }
  }

  Future<Map<String, dynamic>?> getActiveRide() async {
    try {
      final response = await apiClient.dio.get('/passenger/rides/active');
      return response.data;
    } catch (e) {
      return null;
    }
  }

  Future<void> cancelRide(String rideId) async {
    try {
      await apiClient.dio.post('/passenger/rides/$rideId/cancel');
    } catch (_) {}
  }

  Future<List<dynamic>> getRideHistory() async {
    try {
      final response = await apiClient.dio.get('/passenger/rides/history');
      return (response.data as List<dynamic>?) ?? [];
    } catch (e) {
      return [];
    }
  }
}

File: apps/driver-app/src/features/passenger/services/passenger-ride.service.ts
import { api } from '../../../api/axios.instance';

export interface CreateRidePayload {
  pickup: {
    lat: number;
    lng: number;
    address: string;
  };
  destination: {
    lat: number;
    lng: number;
    address: string;
  };
  serviceType: 'ECO' | 'COURSE_PLUS' | 'CONFORT' | 'TAXI' | 'MOTO' | 'CARGO' | 'COMFORT';
  estimatedFareMAD: number;
}

export const passengerRideService = {
  createRide: async (payload: CreateRidePayload) => {
    const response = await api.post('/passenger/rides', {
      pickupLat: payload.pickup.lat,
      pickupLng: payload.pickup.lng,
      pickupAddress: payload.pickup.address || 'Pickup Location',
      dropoffLat: payload.destination.lat,
      dropoffLng: payload.destination.lng,
      dropoffAddress: payload.destination.address || 'Destination Location',
      serviceType: payload.serviceType,
      offeredPrice: payload.estimatedFareMAD,
    });
    return response.data;
  },

  cancelRide: async (rideId: string) => {
    const response = await api.post(`/passenger/rides/${rideId}/cancel`);
    return response.data;
  },

  getActiveRide: async () => {
    try {
      const response = await api.get('/passenger/rides/active');
      return response.data;
    } catch (_) {
      return null;
    }
  },
};
